File: music-commons-core/src/test/java/net/transgressoft/commons/music/playlist/MutablePlaylistDirectoryTest.java
package net.transgressoft.commons.music.playlist;

import net.transgressoft.commons.music.MusicLibraryTestBase;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.util.Collections;

import static com.google.common.truth.Truth.assertThat;

class MutablePlaylistDirectoryTest extends MusicLibraryTestBase {


    @Test
    @DisplayName("Mutable audio directory attributes and operations")
    void additionAndDeletionOfPlaylistsTest() {
        var directory1 = new MutablePlaylistDirectory<>(1, "Directory1");
        assertThat(directory1.isDirectory()).isTrue();
        assertThat(directory1.descendantPlaylists()).isEmpty();
        assertThat(directory1.toString()).isEqualTo("MutablePlaylistDirectory{id=1, name=Directory1, descendantPlaylists=0, audioItems=0}");

        var audioItems = createTestAudioItemsSet(5);
        var p1 = new MutablePlaylist<>(10, "p1", audioItems);
        var p2 = new MutablePlaylist<>(11, "p2");
        var d1 = new MutablePlaylistDirectory<>(12, "d1", Collections.singletonList(createTestAudioItem("One")));

        directory1.addPlaylists(p1, p2, d1);
        assertThat(directory1.descendantPlaylists()).hasSize(3);
        assertThat(directory1.containsPlaylist(d1)).isTrue();

        d1.clearAudioItems();
        assertThat(d1.audioItems()).isEmpty();

        var directory2 = new MutablePlaylistDirectory<>(1, "Directory1");
        assertThat(directory1).isEqualTo(directory2);
    }
}

File: music-commons-core/src/test/kotlin/net/transgressoft/commons/music/playlist/MutablePlaylistDirectoryTest.kt
package net.transgressoft.commons.music.playlist

import com.google.common.truth.Truth.assertThat
import net.transgressoft.commons.music.MusicLibraryTestBase
import net.transgressoft.commons.music.audio.AudioItem
import org.junit.jupiter.api.Test

internal class MutablePlaylistDirectoryTest : MusicLibraryTestBase() {

    @Test
    fun `Mutable audio directory attributes and operations`() {
        val directory1 = MutablePlaylist<AudioItem>(1, true,"Directory1")

        assertThat(directory1.isDirectory).isTrue()
        assertThat(directory1.playlists).isEmpty()
        assertThat(directory1.toString()).isEqualTo("ImmutablePlaylist(id=1, isDirectory=true, name='Directory1', audioItems=[], playlists=[])")

        val audioItems = createTestAudioItemsSet(5)
        val p1 = MutablePlaylist(10, true, "p1", audioItems = audioItems.toMutableList())
        val p2 = MutablePlaylist<AudioItem>(11, true, "p2")
        val d1 = MutablePlaylist(12, true, "d1", audioItems = mutableListOf(createTestAudioItem("One")))

        directory1.playlists.addAll(listOf(p1, p2, d1))
        assertThat(directory1.playlists).hasSize(3)
        assertThat(directory1.playlists.contains(d1)).isTrue()

        d1.audioItems.clear()
        assertThat(d1.audioItems).isEmpty()

        val directory2 = MutablePlaylist<AudioItem>(1, true, "Directory1")

        assertThat(directory1).isEqualTo(directory2)
    }
}